File: tests/combi/matrix_intro_mops.ml
open Optitrust
open Target
open Ast


let _ = Run.doc_script_cpp (fun _ ->

  !! Matrix.intro_mops (var "nbCells") [cVarDef "deposit"];
  
)

"
#include <stdlib.h>
int main() {
  const int nbCells = 10;
  double* deposit = (double*) malloc (nbCells * sizeof(int));
  for (int idCell = 0; idCell < nbCells; idCell++) {
    deposit[idCell] += 1.0;
  }
  return 0;
}
"

let _ = Run.script_cpp (fun _ ->

  !! Matrix.intro_mops (trm_var "N") [cVarDef "p"];
  !! Matrix.intro_mops (trm_var "N") [cVarDef "q"];

)

File: tests/combi/matrix_intro_mops.cpp
#include <stdlib.h>
int main () {

  const int N = 5;
  int* p = (int*) calloc (N, sizeof(int));
  for (int i = 1; i < 5; i++){
    int a = p[i-1];
    p[i] = i;
  }
  int* q = (int*) malloc (N * sizeof(int));
  for (int i = 0; i < 5; i++){
    q[i] = i;
  }

  return 0;
}